src/modules/orbitdb/db-index.js

```diff
@@ -0,0 +1,63 @@
+const IPFSUtils = require('../../utils/ipfs');
+
+const version = 0;
+
+class ChluIndex {
+    constructor(){
+        this._index = {
+            reviewRecords: [],
+            updates: {}
+        };
+    }
+
+    /**
+     * Called by OrbitDB to update the index 
+     */
+    updateIndex(oplog) {
+        oplog.values
+            .slice()
+            .forEach(item => {
+                // Skip operations from a different chlu store version
+                if (item.payload.version === version) {
+                    // TODO: check update validity, RR validity if possible
+                    if (item.payload.op === operations.ADD_REVIEW_RECORD && IPFSUtils.isValidMultihash(item.payload.multihash)) {
+                        // Do not create duplicates
+                        if (this._index.reviewRecords.indexOf(item.payload.multihash) < 0) {
+                            this._index.reviewRecords.splice(0, 0, item.payload.multihash);
+                        }
+                    } else if (item.payload.op === operations.UPDATE_REVIEW_RECORD && IPFSUtils.isValidMultihash(item.payload.multihash) && IPFSUtils.isValidMultihash(item.payload.previousVersionMultihash)) {
+                        // TODO: check that whatever is being updated is in the DB
+                        // the multihash to be updated might already be an update. Be careful!
+                        const from = this.getLatestReviewRecordUpdate(item.payload.previousVersionMultihash);
+                        this._index.updates[from] = item.payload.multihash;
+                    }
+                }
+            });
+    }
+
+    getLatestReviewRecordUpdate(multihash, stack = [multihash]) {
+        IPFSUtils.validateMultihash(multihash);
+        let next;
+        next = this._index.updates[multihash];
+        if (IPFSUtils.isValidMultihash(next) && stack.indexOf(next) === -1) {
+            // One next iteration
+            return this.getLatestReviewRecordUpdate(next, stack.concat(next));
+        } else {
+            // End of search, return latest valid data
+            return stack[stack.length-1];
+        }
+    }
+
+    getReviewRecordList() {
```

Comment:
Will this result in updates being missed? So when I ask for review record that has been updated once, will this call result in returning just the original review as part of this list?

Reply:
OK. I see. You then call getLatestReviewRecordUpdate for the original RR

Reply:
Yeah, I thought it's better to only keep a list of first-version reviews, then follow the pointers to new versions only if explicitly requested